--- origsrc/libass-0.9.11/libass/Makefile.am	2010-10-20 01:26:35.921717300 -0500
+++ src/libass-0.9.11/libass/Makefile.am	2010-10-20 01:27:24.443867100 -0500
@@ -12,6 +12,7 @@ libass_la_SOURCES = ass.c ass_cache.c as
                     ass_drawing.h ass_cache_template.h ass_render.h \
                     ass_parse.c ass_parse.h ass_render_api.c ass_strtod.c
 libass_la_LDFLAGS = -version-info $(LIBASS_LT_CURRENT):$(LIBASS_LT_REVISION):$(LIBASS_LT_AGE)
+libass_la_LDFLAGS += -no-undefined
 libass_la_LDFLAGS += -export-symbols $(srcdir)/libass.sym
 
 assheadersdir = $(includedir)/ass
